api/database/migrations/2022_09_06_032007_generated_column_priority.php

```diff
@@ -14,13 +14,13 @@ class GeneratedColumnPriority extends Migration
      */
     public function up()
     {
-        DB::statement("ALTER TABLE users ADD COLUMN priority_status VARCHAR(50)
+        DB::statement("ALTER TABLE users ADD COLUMN priority_status INT
         GENERATED ALWAYS AS
         (case
-        when has_priority_entitlement = TRUE then 'PRIORITY'
-        when armed_forces_status = 'VETERAN' then 'VETERAN'
-        when citizenship = 'CITIZEN' OR citizenship = 'PERMANENT_RESIDENT' then 'CITIZEN_OR_PR'
-        else 'OTHER'
+        when has_priority_entitlement = TRUE then 1
```

Comment:
Might it be worth it to have values staggered so that we can add types in the future in between?
```
has_priority_entitlement = 1
armed_forces_status = 10
citizenship = 'CITIZEN' OR citizenship = 'PERMANENT_RESIDENT' = 20
else = 30
```

Reply:
https://github.com/GCTC-NTGC/gc-digital-talent/pull/3811/commits/b826ec992e16f4b98d58027658ebccc6ae30bbcc